woocommerce/woocommerce
# Deprecate and create new coming soon block version

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





A follow-up PR for https://github.com/woocommerce/woocommerce/pull/49749#issuecomment-2244139884

This PR deprecates the existing Coming Soon block and creates a new version of the block with the following changes:

- Use scss for styling, instead of inline styles in JSX/PHP. This will make it easier to maintain and update the styles. Also, it keeps the `save()` render clean so we don't need to depreciate the block every time we change the styles.
- Add className `woocommerce-coming-soon-store-only`, `woocommerce-coming-soon-entire-site` to the block wrapper to make it easier to style the block.

We can change `Edit()` function anytime without worrying about the making block incompatible with the previous version. We only need to ensure that `save()` function returns the same output as the previous version. Otherwise, we need to deprecate the block and create a new version of the block.

Reference: https://developer.wordpress.org/block-editor/reference-guides/block-api/block-deprecation/




### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:


### Test site already has customization before switching to the new block version

1. Set up a JN site using woocommerce stable version (9.1) and WC beta tester plugin
2. Skip Core Profiler
3. Go to `Appearance > Editor > Templates > Page: Coming Soon`
4. Edit the text in the middle and save
5. Go to `WooCommerce > Live Branches` and Install & activate this branch
6. Go to `Appearance > Editor > Templates > Page: Coming Soon`
7. Confirm that the page is rendered as previously without any invalid block notices
8. Go to /wp-admin/site-editor.php?postType=wp_template
9. Search for the Coming Soon template
10. Observe the preview should be displayed correctly
11. Go to site front-end and confirm that the page is rendered as expected without errors

### Test new site

1. Set up a JN site using this branch
2. Go to `Appearance > Editor > Templates > Page: Coming Soon`
3. Skip Core Profiler
4. Confirm that the page is rendered as expected without errors
5. Go to site front-end and confirm that the page is rendered as expected without errors
6. Go to wp-admin/admin.php?page=wc-settings&tab=site-visibility
7. Turn on `Restrict to store pages only` option
8. Go to site front-end > shop page, confirm that the page is rendered as expected without errors
9. Go to `Appearance > Editor > Templates > Page: Coming Soon`
10. Confirm that the page is rendered as expected without errors
11. Select the parent block and confirm that you can change the background color

Entire Site:

![Screenshot 2024-07-23 at 15 57 02](https://github.com/user-attachments/assets/509e3ad0-219a-490e-9161-db9a8e1cf19a)

Store Only:

![Screenshot 2024-07-23 at 15 56 36](https://github.com/user-attachments/assets/0334c6cd-f8b2-440a-9f83-75a46c54f3bb)

![Screenshot 2024-07-23 at 16 01 05](https://github.com/user-attachments/assets/f6956cbb-5e96-45ad-bedd-74d9bed985a4)





### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

Deprecate and create new coming soon block version

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
